src/generator-attribute-manager.cs

```diff
@@ -83,8 +85,56 @@ static Type ConvertType (System.Type type, ICustomAttributeProvider provider)
 		return rv;
 	}
 
-	static System.Attribute CreateAttributeInstance (CustomAttributeData attribute, ICustomAttributeProvider provider)
+
+	static IEnumerable<System.Attribute> ConvertOldAttributes (CustomAttributeData attribute)
+	{
+		switch (attribute.AttributeType.Namespace) {
+		case "MonoTouch.ObjCRuntime":
+		case "ObjCRuntime":
+		case "MonoTouch.ObjCRuntime.Extensions":
+		case "ObjCRuntime.Extensions":
+			break;
+		default:
+			return Enumerable.Empty<System.Attribute> ();
+		}
+
+		switch (attribute.AttributeType.Name) {
+		case "SinceAttribute":
+		case "iOSAttribute":
+			return AttributeConversionManager.ConvertPlatformAttribute (attribute, 2 /* PlatformName.iOS */).Yield ();
```

Comment:
Would it be better to cast these from the PlatformName enums instead of hard-coding the numbers, simply to make it harder to accidentally put in the wrong value if changes need to be made in the future? (And so one wouldn't have to look up what values they need to use from the enum if they have to make changes).